app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -0,0 +1,126 @@
+package org.oppia.app.home.topiclist
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+import androidx.databinding.library.baseAdapters.BR
+import org.oppia.app.R
+import org.oppia.app.databinding.PromotedStoryCardBinding
+import org.oppia.app.databinding.TopicSummaryViewBinding
+import org.oppia.app.databinding.WelcomeBinding
+import org.oppia.app.home.UserAppHistoryViewModel
+import org.oppia.app.model.TopicSummary
+
+private const val VIEW_TYPE_WELCOME_MESSAGE = 1
+private const val VIEW_TYPE_PROMOTED_STORY = 2
+private const val VIEW_TYPE_TOPIC_LIST = 3
+
+/** Adapter to inflate different items/views inside [RecyclerView]. The itemList consists of various ViewModels. */
+class TopicListAdapter(
+  private val itemList: MutableList<Any>,
+  private val topicSummaryClickListener: TopicSummaryClickListener
+) :
+  RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      // TODO(#249): Generalize this binding to make adding future items easier.
```

Comment:
I think this falls more under #216 since using the bindable adapter makes things pretty generic, especially when combined with an enum (see #270 in StateFragmentPresenter for how this looks when it's all composed).

Reply:
Done. And it makes sense.